components/popover/stories/template.js

```diff
@@ -42,93 +42,97 @@ export const Template = ({
+								: "- var(--spectrum-popover-animation-distance)";
+						}
+					} else if (position.includes("right")) {
+						if (textDir == 'rtl') {
+							x = rect.left - popWidth;
+							xOffset = withTip
+								? "- ((var(--spectrum-popover-pointer-width) / 2) + var(--spectrum-popover-animation-distance) - 2px)"
+								: "- var(--spectrum-popover-animation-distance)";
+						} else {
+							x = rect.right;
+							xOffset = withTip ? "+ 0px" : "+ var(--spectrum-popover-animation-distance)";
+						}
 					}
-				}
-
-				if (x) transforms.push(`translateX(calc(var(--flow-direction) * calc(${parseInt(x, 10)}px ${xOffset})))`);
-				if (y) transforms.push(`translateY(calc(${y}px ${yOffset}))`);
 
-				// Add start and end styles
-				if (position === "top-start" || position === "bottom-start") {
-					additionalStyles["inset-inline-start"] = "calc(" + (popWidth / 2) + "px - var(--spectrum-popover-pointer-edge-offset))";
-				} else if (position === "top-end" || position === "bottom-end") {
-					additionalStyles["inset-inline-start"] = "calc(-1 *" + (popWidth / 2) + "px + var(--spectrum-popover-pointer-edge-offset))";
-				} else if (position === "left-top" || position === "right-top") {
-					additionalStyles["inset-block-start"] = "calc(" + (popHeight / 2) + "px - var(--spectrum-popover-pointer-edge-offset))";
-				} else if (position === "left-bottom" || position === "right-bottom") {
-					additionalStyles["inset-block-start"] = "calc(-1 *" + (popHeight / 2) + "px + var(--spectrum-popover-pointer-edge-offset))";
-				}
+					if (x) transforms.push(`translateX(calc(var(--flow-direction) * calc(${parseInt(x, 10)}px ${xOffset})))`);
+					if (y) transforms.push(`translateY(calc(${y}px ${yOffset}))`);
 
-				updateArgs({
-					isOpen: !isOpen,
-					customStyles: {
-						...customStyles,
-						transform: transforms.join(" "),
-						...additionalStyles,
+					// Add start and end styles
+					if (position === "top-start" || position === "bottom-start") {
+						additionalStyles["inset-inline-start"] = "calc(" + (popWidth / 2) + "px - var(--spectrum-popover-pointer-edge-offset))";
+					} else if (position === "top-end" || position === "bottom-end") {
+						additionalStyles["inset-inline-start"] = "calc(-1 *" + (popWidth / 2) + "px + var(--spectrum-popover-pointer-edge-offset))";
+					} else if (position === "left-top" || position === "right-top") {
+						additionalStyles["inset-block-start"] = "calc(" + (popHeight / 2) + "px - var(--spectrum-popover-pointer-edge-offset))";
+					} else if (position === "left-bottom" || position === "right-bottom") {
+						additionalStyles["inset-block-start"] = "calc(-1 *" + (popHeight / 2) + "px + var(--spectrum-popover-pointer-edge-offset))";
 					}
-				});
-			}, 100)
-			}
+
+					updateArgs({
+						isOpen: !isOpen,
+						customStyles: {
+							...customStyles,
+							transform: transforms.join(" "),
+							...additionalStyles,
+						}
+					});
+				}, 100)
+				}
```

Comment:
I didn't actually change any of this code, I just indented it under `!nestedPopover &&`, which I did add.